src/dev/typescript/build_ts_refs_cli.ts

```diff
@@ -30,74 +30,80 @@ const isTypeFailure = (error: any) =>
 
 export async function runBuildRefsCli() {
   run(
-    async ({ log, flags, procRunner }) => {
-      if (process.env.BUILD_TS_REFS_DISABLE === 'true' && !flags.force) {
-        log.info(
-          'Building ts refs is disabled because the BUILD_TS_REFS_DISABLE environment variable is set to "true". Pass `--force` to run the build anyway.'
-        );
-        return;
-      }
-
-      // if the tsconfig.refs.json file is not self-managed then make sure it has
-      // a reference to every composite project in the repo
-      await updateRootRefsConfig(log);
-
-      // load all the projects referenced from the root refs config deeply, so we know all
-      // the ts projects we are going to be cleaning or populating with caches
-      const projects = Project.load(
-        ROOT_REFS_CONFIG_PATH,
-        {},
-        {
-          skipConfigValidation: true,
-        }
-      ).getProjectsDeep(PROJECT_CACHE);
-
+    async ({ log, flags, procRunner, statsMeta }) => {
+      const enabled = process.env.BUILD_TS_REFS_DISABLE === 'true' && !flags.force;
       const cacheEnabled = process.env.BUILD_TS_REFS_CACHE_ENABLE !== 'false' && !!flags.cache;
       const doCapture = process.env.BUILD_TS_REFS_CACHE_CAPTURE === 'true';
       const doClean = !!flags.clean || doCapture;
       const doInitCache = cacheEnabled && !doCapture;
 
-      if (doClean) {
-        log.info('deleting', projects.outDirs.length, 'ts output directories');
-        await concurrentMap(100, projects.outDirs, (outDir) => del(outDir));
-      }
+      statsMeta.set('buildTsRefsEnabled', enabled);
+      statsMeta.set('statsMeta.buildTsRefsCacheEnabled', cacheEnabled);
+      statsMeta.set('statsMeta.buildTsRefsDoCapture', doCapture);
+      statsMeta.set('statsMeta.buildTsRefsDoClean', doClean);
+      statsMeta.set('statsMeta.buildTsRefsDoInitCache', doInitCache);
 
-      let outputCache;
-      if (cacheEnabled) {
-        outputCache = await RefOutputCache.create({
-          log,
-          projects,
-          repoRoot: REPO_ROOT,
-          workingDir: CACHE_WORKING_DIR,
-          upstreamUrl: 'https://github.com/elastic/kibana.git',
-        });
-      }
+      if (enabled) {
```

Comment:
Mind using an early `return` instead of wrapping tons of lines in an `else` branch?

Reply:
I still want to collect metrics on disabled builds.

Reply:
Actually, nm - this works now that I have it in run.

Reply:
Will restore to what it was.

Reply:
`return;` will just end the `async` function, it won't prevent `run()` from sending metrics.